tizen-api/src/tizen-api-pipeline.c

```diff
@@ -153,6 +155,31 @@ cb_sink_event (GstElement * e, GstBuffer * b, gpointer data)
   return;
 }
 
+/**
+ * @brief Private function for nns_pipeline_destroy, cleaning up nodes in namednodes
+ */
+static void
+cleanup_node (gpointer data)
+{
+  element *e = data;
+  g_mutex_lock (&e->lock);
+  g_free (e->name);
+  if (e->src)
+    gst_object_unref (e->src);
+  if (e->sink)
+    gst_object_unref (e->sink);
+
+  /** @todo CRITICAL. Stop the handle callbacks if they are running/ready */
+  if (e->handles)
+    g_list_free (e->handles);
```

Comment:
if each handles are dynamically allocated, it should be freed with g_list_free_full.

Reply:
Thanks! Will update soon.